hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/error/ErrorBuilder.java

```diff
@@ -10,78 +12,204 @@
 	
 	private static String ack= "ACK";
 	
-	private static String version = "2.4\n";
+	private static String msh= "MSH";
 
 	private static String msa = "MSA|AR|";
-
-	private static String seperator3 = "|||";
+	
+	private static String unknown = "UNKNOWN";
+	
+	private static String sendingOrReceivingApp = "HNCLIENT";
+	
+	private  static String double_backslash = "\\"; // For using specific string in regex mathces
+	
+	private  static String delimiter = "|";
 
 	private static Timestamp timestamp = new Timestamp(System.currentTimeMillis());
 
-	private static String processingDomain = "D";
-
 	private static String sendingFacility;
 
-	private static String recievingFacility;
-
+	private static String receivingFacility;
+	
 	@Deprecated
 	public static String generateError(Integer errorCode, String v2Msg) {
 
 		String errCode = ErrorCodes.retrieveEnumByValue(errorCode);
-		return buildHTTPErrorMessage(errCode, v2Msg);
+		return buildErrorMessage(errCode, v2Msg);
 
 	}
 
-	public static String buildHTTPErrorMessage(String errMsg,String v2Msg) {
-		return buildMSH(v2Msg) +buildMSA(errMsg);
+	/**
+	 * @param v2Msg
+	 * @param errMsg
+	 * @return
+	 */
+	public static String buildErrorMessage(String v2Msg,String errMsg) {
+		return buildMSH(v2Msg) +buildMSA(v2Msg,errMsg);
 	}
 
-	public static String buildDefaultErrorMessage(String errMsg) {
-		return buildMSH(HL7Error_Default) + buildMSA(errMsg) ;
+	private static String buildMSA(String v2Msg, String errMsg) {
+		if(StringUtil.isNullOrEmpty(v2Msg))
+			v2Msg= HL7Error_Default;
```

Comment:
It's best to avoid braceless if statements as they are error prone if someone wants to add additional logic to the if statement it can end up outside the if statement.